### PR TITLE
[CDAP-15053] Store CDAP version in Elasticsearch index metadata

### PR DESCRIPTION
**Pull request by anew**
Stores the CDAP version in the Elasticsearch index metadata. Elastic's way of doing this is (interestingly enough) to create a mapping for the special field _meta. This mapping is ignored by Elastic, but can be retrieved as part of a GetMappings request.

### CODE REVIEW

**Comment by anew** [comment]
DUT: https://builds.cask.co/browse/CDAP-DUT6920
DBV: https://builds.cask.co/browse/CDAP-DBV31
ITM: https://builds.cask.co/browse/IT-ITM-149

File: cdap-elastic/src/main/resources/index.mapping.json [hunk 1]
```diff
@@ -17,6 +17,9 @@
 // Note: Strictly speaking, JSON does not allow comments, but GSON and Elasticsearch appear to be tolerant.
 // This file directly corresponds to the structure of co.cask.cdap.metadata.elastic.MetadataDocument.
 {
+  "_meta": {
+    "cdap_version": "CDAP_VERSION"
```
Comment (chtyim) [inline cdap-elastic/src/main/resources/index.mapping.json]: For file in `resources`, you can actually relies on maven to do the auto substitution for you. Basically in here, you can have:

```java
"cdap_version": "${project.version}"
```

and then in the pom.xml, has
```
  <build>
    <resources>
      <resource>
        <directory>src/main/resources</directory>
        <filtering>true</filtering>
      </resource>
    </resources>
   .....
  </build>
```

You don't need any code to do the replacement. This is how we do for the ProjectInfo version:

https://github.com/cdapio/cdap/blob/develop/cdap-common/src/main/resources/build.properties
https://github.com/cdapio/cdap/blob/develop/cdap-common/pom.xml#L199

Comment (anew) [reply cdap-elastic/src/main/resources/index.mapping.json]: That makes sense. However, on request by @sreevatsanraman , I am now also storing a "metadata version" and a checksum of the mappings file. That will help avoid migration when the CDAP version changes but the metadata is compatible. 

Using resource filtering to replace the cdap version would now change the checksum of this resource on every build, which is not what we want. So I prefer to make that replacement programmatically.

**Review by sreevatsanraman — APPROVED** [review]

**Comment by anew** [comment]
new ITM has passed: https://builds.cask.co/browse/IT-ITM-154
squashing and merging

**Pull request merged**